src/main/java/com/github/psambit9791/jdsp/transform/DiscreteFourier.java

```diff
@@ -60,61 +64,36 @@ public void dft() {
     // About plotting, please refer here: https://stackoverflow.com/a/25735274
 
     /**
-     * Returns the absolute value of the discrete fourier transformed sequence
+     * Returns the absolute value (amplitude) of the discrete fourier transformed sequence
      * @param onlyPositive Set to True if non-mirrored output is required
      * @throws java.lang.ExceptionInInitializerError if called before executing dft() method
      * @return double[] The absolute DFT output
      */
     public double[] returnAbsolute(boolean onlyPositive) throws ExceptionInInitializerError{
-        if (this.output == null) {
-            throw new ExceptionInInitializerError("Execute dft() function before returning result");
-        }
-        double[] dftout;
+        Complex[] dftout = returnComplex(onlyPositive);
+        return Arrays.stream(dftout).mapToDouble(Complex::abs).toArray();
+    }
 
-        if (onlyPositive) {
-            int numBins = this.output.length/2+1;
-            dftout = new double[numBins];
-            for (int i=0; i<dftout.length; i++) {
-                dftout[i] = this.output[i].abs();
-            }
-        }
-        else{
-            dftout = new double[this.output.length];
-            for (int i=0; i<dftout.length; i++) {
-                dftout[i] = this.output[i].abs();
-            }
-        }
-        return dftout;
+    /**
+     * Returns the argument value (phase) of the discrete fourier transformed sequence
+     * @param onlyPositive Set to True if non-mirrored output is required
+     * @throws java.lang.ExceptionInInitializerError if called before executing dft() method
+     * @return double[] The argument DFT output
+     */
+    public double[] returnArgument(boolean onlyPositive) throws ExceptionInInitializerError{
```

Comment:
Same as before. Might be useful to make this returnPhase()

Reply:
+1